vzstorage-pd/vendor/github.com/virtuozzo/goploop-cli/ploopvol/ploop-volume.go

```diff
@@ -0,0 +1,65 @@
+package ploopvol
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+)
+
+var (
+	ploopVolumeBinary = "/usr/sbin/ploop-volume"
+)
+
+type PloopVolume struct {
+	Path string
+}
+
+func (pv *PloopVolume) Create(size uint64, image string) error {
+	args := []string{"create", "-s", fmt.Sprintf("%v", size)}
+	if image != "" {
+		args = append(args, "--image", image)
+	}
+	args = append(args, pv.Path)
+	cmd := exec.Command(ploopVolumeBinary, args...)
+	return cmd.Run()
+}
+
+func (pv *PloopVolume) Snapshot(dst string) error {
+	if dst == "" {
+		return fmt.Errorf("Bad destination path!")
+	}
+	args := []string{"snapshot", pv.Path, dst}
+	cmd := exec.Command(ploopVolumeBinary, args...)
+	return cmd.Run()
+}
+
+func (pv *PloopVolume) Clone(dst string) error {
+	if dst == "" {
+		return fmt.Errorf("Bad destination path!")
+	}
+	backupPath := path.Clean(dst) + ".backup"
+
+	if _, err := os.Stat(pv.Path); !os.IsNotExist(err) {
+		if err = os.Rename(pv.Path, backupPath); err != nil {
```

Comment:
I don't like this game with backupPath(). Clone() has to fail if a destination path already exists.